### PR TITLE
[CoudFormation] Add 1-click template to deploy a proxy environment for ParallelCluster.

### PR DESCRIPTION
**Pull request by gmarciani**
### Description of changes
Add 1-click template to deploy a proxy environment for ParallelCluster.

### Tests
* Template deployed successfully
* Deployed successfully a pcluster using the deployed proxy
* Verified injecting a manual error in the proxy, that the template fails the deployment is the proxy setup does not work properly.

### References
* Link to impacted open issues.
* Link to related PRs in other packages (i.e. cookbook, node).
* Link to documentation useful to understand the changes.

### Checklist
- Make sure you are pointing to **the right branch**.
- If you're creating a patch for a branch other than `develop` add the branch name as prefix in the PR title (e\.g\. `[release-3.6]`).
- Check all commits' messages are clear, describing what and why vs how.
- Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cloudformation/proxy/proxy.yaml [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+  # SECURITY GROUPS
+  ProxySecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: Security Group for the proxy node
+      SecurityGroupIngress:
+        - CidrIp: !GetAtt PrivateSubnet.CidrBlock
+          Description: Allow all inbound traffic from private subnet
+          IpProtocol: -1
+        - SourcePrefixListId: pl-01a74268 # pl-60b85b09 - com.amazonaws.firewall.regional-corp-only
```
Comment (himani2411) [inline cloudformation/proxy/proxy.yaml]: Is this a customer facing template or just for internal testing?

Comment (himani2411) [reply cloudformation/proxy/proxy.yaml]: IF its customer facing then I believe we shouldnt be adding the Prefix list

Comment (gmarciani) [reply cloudformation/proxy/proxy.yaml]: Great catch: it's a left over from my tests. Removing them

Comment (gmarciani) [reply cloudformation/proxy/proxy.yaml]: Removed in next rev

**Review by himani2411 — APPROVED** [review]

**Comment by himani2411** [comment]
LGTM!

**Pull request merged**